rfd/XXXX-teleport-release-engineering.md

```diff
@@ -0,0 +1,423 @@
+                drone->>-releasesvr: Register Releaseable file artifacts
+
+            else On GitHub Actions
+                drone->>+gha: Trigger $WORKFLOW for vX.Y.Z
+                dronestorage->>+gha: Read intermediate file artifacts
+                Note over gha: Build activities happen
+                gha->>dronestorage: Write intermediate file artifacts
+                gha->>intermediate_repos: Write intermediate non-file artifacts
+                gha->>-releasesvr: Register Releaseable file Artifacts
+                gha->>drone: Workflow complete
+            end 
+        end
+        drone->>-release_engineer: Build vX.Y.Z complete
+    end
+
+    rect rgb(250, 250, 250)
+        Note over release_engineer: Promotion Phase
+
+        release_engineer->>+drone: Promote build vX.Y.Z
+        
+        dronestorage->>drone: Gather file artifacts for vX.Y.Z
+        drone->>release_storage: Move file artifacts for vX.Y.Z into release bucket
+        
+        intermediate_repos->>drone: Gather non-file artifacts for vX.Y.Z
+        drone->>release_repos: Move non-file artifacts into release repositories (Container Images, AMIs, Deb Packages)
+    end
+
+    rect rgb(250, 250, 250)
+        Note over release_engineer: Publishing Phase
+        release_engineer->>github: Create Github Release
+        github->>releasesvr: New Release vX.Y.Z (Actually polling from Release Server)
+        Note over releasesvr: Copy registered artifacts<br/>from staging area to<br/>Production CDN
+    end
+
+    rect rgb(250, 250, 250)
+        Note over customer: Later on...
+        customer->>website: I want to download Teleport
+        website->>releasesvr: What are the artifacts for vX.Y.Z
+        releasesvr->>website: List of artifacts for vX.Y.Z with download links to CDN
+    end
+```
+
+> Note: As far as I can figure out, the contents of the `Release Artifact
+> Bucket` are never actually read from anything. Release artifacts are served
+> from the `Release API Server`'s CloudFront-backed CDN. 
+>
+> This bucket _may_ be required for build traceability, but could just as easily
+> exist for legacy reasons (i.e. no-one wanted to delete it incase it broke
+> something).
+
+#### **Infrastructure**
+
+Drone is entirely self hosted, we (Teleport) need to manage it ourselves. 
+
+GHA, on the other hand, is largely managed for us by GitHub, although we
+do use pools of ephemeral, self-hosted runners for specific tasks. 
+
+We currently manage, as part of our CI/CD/Release system:
+ - AWS EKS cluster running `Drone` (see `./drone` in the `ops` repository)
+ - **Drone runners:**
+    - 3x (one per actively supported major Teleport Version) AWS EC2 AMD64
+      instances for Windows Builds
+    - Intel MacMini in Macstadium
+ - **Self-Hosted GHA Runners**
+    - Ephemeral AWS EC2 arm64 boxes for arm64 linux builds. Pool has a max of 10
+    instances
+    - Ephemeral EC2 instance with a persistent volume for building `deb` 
+    and `rpm` packages.
```

Comment:
@fheinecke, @camscale - I'm pretty sure this is no longer valid. What are we doing now?